### DISCUSSION TITLE
Shift+drag in the costume editor to pan

### DISCUSSION THREAD

**Discussion by arav-tya** [ScratchAddons/ScratchAddons#5358] [Ideas]
I find it annoying how I have to adjust the little bars in the corners when I move around the costume editor. Maybe holding shift + drag with the select tool should allow you to pan it freely?

**Comment by BroJac5246** [comment]
I didn't even realize there's a need for this– I just use my trackpad.

It's a good idea, though. Does anyone know if `Shift + Drag` conflicts with anything?

**Reply by Samq64** [reply]
It seems to bring up the devtools block search.

**Reply by BroJac5246** [reply]
> It seems to bring up the devtools block search.

Even in the paint editor? If so, that's a bug that needs to be fixed.

**Reply by mxmou** [reply]
> > It seems to bring up the devtools block search.
> 
> Even in the paint editor? If so, that's a bug that needs to be fixed.

Only in the code editor.

**Reply by Samq64** [reply]
I missread. Yeah this would be useful, what about middle-click + drag like TurboWarp does?

**Reply by BroJac5246** [reply]
> I missread. Yeah this would be useful, what about middle-click + drag like TurboWarp does?

I was about to point out that not everyone has a scroll wheel, but if they don't then they have a trackpad and can use it to scroll. That means that `middle-click + drag` is a good choice, since only those who need it could use it.

I don't have a mouse, so now I'm wondering: what happens if you press the scroll wheel while in the paint editor? Can you use that to scroll around?

**Reply by arav-tya** [reply]
middle click + drag is a better idea actually

**Reply by arav-tya** [reply]
pressing midde-click in the costume editor doesnt do anything

**Reply by BroJac5246** [reply]
> pressing midde-click in the costume editor doesnt do anything

Could we just add that? That seems simpler and more logical.

**Reply by Secret-chest** [reply]
> > pressing midde-click in the costume editor doesnt do anything
> 
> Could we just add that? That seems simpler and more logical.

And it's standard.
